enterprise/server/remote_execution/runner/runner.go

```diff
@@ -540,22 +526,23 @@ func (r *commandRunner) cleanupCIRunner(ctx context.Context) error {
 	return res.Error
 }
 
-type ContainerProvider func(ctx context.Context, props *platform.Properties, st *repb.ScheduledTask, state *rnpb.RunnerState, workDir string) (*container.TracedCommandContainer, error)
+type ContainerProvider interface {
+	NewContainer(context.Context, *platform.Properties, *repb.ScheduledTask, *rnpb.RunnerState, string) (container.CommandContainer, error)
+}
 
 type PoolOptions struct {
 	// ContainerProvider is an optional implementation overriding
 	// newContainerImpl.
-	ContainerProvider ContainerProvider
+	CtrProvider ContainerProvider
```

Comment:
the old name `ContainerProvider` seemed better to me - I think generally we avoid abbreviations for public fields